core/src/main/java/ch/cyberduck/core/vault/DecryptingListProgressListener.java

```diff
@@ -54,6 +57,9 @@ public void visit(final AttributedList<Path> list, final int index, final Path f
                 }
                 list.set(index, vault.decrypt(session, f).withAttributes(new PathAttributes(f.attributes()).withVersions(versions)));
             }
+            if (find != null && list.get(index).isDirectory() && !find.find(vault.encrypt(session, list.get(index)))) {
```

Comment:
This will cause performance issues with large directory listings.

Reply:
Is there a better way to do this? Any directory that fails this check does not actually exist, and problems will appear later when you try to e.g. delete it.

Reply:
As this is the exception it would probably be a better approach to just ignore failures for missing files on delete silently.

Reply:
This is really a problem with listings and can still appear if you don't try to delete anything. If the directory shows up in a listing even though it doesn't exist, almost anything that interacts with it will fail, including listing - not just deleting.